File: plugins/devkit/devkit-java-tests/testData/threadingModelHelper/BothAnnotationAndAssertion.java
import com.intellij.util.concurrency.ThreadingAssertions;
import com.intellij.util.concurrency.annotations.RequiresWriteLock;

class BothAnnotationAndAssertion {
    @RequiresWriteLock
    void testMethod() {
        ThreadingAssertions.assertBackgroundThread();
    }
}

File: plugins/devkit/devkit-kotlin-tests/testData/threadingModelHelper/BothAnnotationAndAssertion.kt
import com.intellij.util.concurrency.annotations.RequiresWriteLock
import com.intellij.util.concurrency.ThreadingAssertions

class BothAnnotationAndAssertion {
  @RequiresWriteLock
  fun testMethod() {
    ThreadingAssertions.assertBackgroundThread()
  }
}
